### DISCUSSION TITLE
Prevent args from changing when using pollingInterval

### DISCUSSION THREAD

**Discussion by l2aelba** [reduxjs/redux-toolkit#4450] [Q&A] [answered]
Example my main page using `pollingInterval` with args `{ limit: 100 }`

```js
export default function MyPage() {
  const { data } = api.endpoints.getSomething.useQuery(
    { limit: 100 },
    { pollingInterval: 10000, skip: false, refetchOnMountOrArgChange: true }
  )
  // …
}
```

And one of my components needs to perform a custom trigger with `{ limit: 1 }` (for performance) like…

```js
export default function MyComponent() {
  const [trigger] = api.endpoints.getSomething.useLazyQuery()
  
  const handleUpdate = () => {
    trigger({ limit: 1 })
  }
  // …
}
```
**Question: 🤔** 
How to keep `MyPage()` polling with args `{ limit: 100 }` ?
Because when `trigger({ limit: 1 })` trigged the args changing to `{ limit: 1 }` in `MyPage()` with `pollingInterval`

**Network logs:**
```js
"https://a.com/something?limit=100"    // pollingInterval
"https://a.com/something?limit=100"    // pollingInterval
"https://a.com/something?limit=100"    // pollingInterval
"https://a.com/something?limit=1"      // Because trigger({ limit: 1 }) triggered
"https://a.com/something?limit=1"      // pollingInterval (args changed)
"https://a.com/something?limit=1"      // pollingInterval
"https://a.com/something?limit=1"      // pollingInterval
// so on…
```

Thanks alot.

**Comment by phryneas** [comment]
That is very weird - `getSomething({ limit: 1 })` and `getSomething({ limit: 100 })` are two separate cache entries with completely separate tracking of polling etc. - they should never cross over.

Could you create a reproduction (e.g. a CodeSandbox) of that behaviour?

**Reply by l2aelba** [reply]
Oh I forgot to tell, I using `merge`, `serializeQueryArgs` and `forceRefetch` for infinite loader…

```js
serializeQueryArgs: ({ endpointName, queryArgs }) => {
  return `${endpointName}-${queryArgs.id}`
},
merge: (currentCache, newItems) => {
  const merged = [...newItems, ...currentCache]
  const map = new Map()
  merged.forEach((item) => map.set(item.id, item))
  const unique = Array.from(map.values())
  unique.sort((a, b) => a.createdAt - b.createdAt)
  return unique
},
forceRefetch({ currentArg, previousArg }) {
  return currentArg !== previousArg
}
```

**Reply by phryneas** [reply, marked as answer]
I believe this one is currently not possible - by using a manual `serializeQueryArgs`, you merge these into semantically one cache entry - and polling will always just "follow" that cache entry :/

**Reply by l2aelba** [reply]
> I believe this one is currently not possible - by using a manual `serializeQueryArgs`, you merge these into semantically one cache entry - and polling will always just "follow" that cache entry :/

Thanks for that, so I will refetch again with the correct args.
